src/nvim/testdir/test_autocmd.vim

```diff
@@ -6,3 +6,20 @@ func Test_vim_did_enter()
   " This script will never reach the main loop, can't check if v:vim_did_enter
   " becomes one.
 endfunc
+
+if !has('timers')
+  finish
+endif
+
+func ExitInsertMode(id)
+  call feedkeys("\<Esc>")
+endfunc
+
+func Test_cursorhold_insert()
+  let g:triggered = 0
+  au CursorHoldI * let g:triggered += 1
+  set updatetime=20
+  call timer_start(100, 'ExitInsertMode')
+  call feedkeys('a', 'x!')
+  call assert_equal(1, g:triggered)
+endfunc
```

Comment:
@Shougo Thank, I get it now. I was confused. I think this is where that check should go:

https://github.com/neovim/neovim/commit/7da7ff7c5c79dd36e053f5b5e86046838445bb8e#diff-ce1f54750dd541783db1cf9a24120ba8R479